scripts/templates/params.hpp.mako

```diff
@@ -365,7 +375,7 @@ template <typename T> inline void serializePtr(std::ostream &os, T *ptr) {
         os << (void *)(ptr) << " (";
         serializePtr(os, *ptr);
         os << ")";
-    } else if constexpr (std::is_void_v<T> || !is_type_complete_v<T>) {
+    } else if constexpr (std::is_void_v<T> || !is_type_complete_v<T> || is_handle_v<T *>) {
```

Comment:
Yeah you're right - I don't think that there are any other incomplete types other than handles.

It does need to be `is_handle_t<T *>` since the template of the function will remove the pointer and `T` will be the incomplete type i.e. `ur_device_handle_t_`

Reply:
Now that `is_type_complete_v`, you can also safely remove its implementation.

Reply:
So the build fails if I remove `is_type_complete_v` because `ur_loader_config_handle_t` is removed from the specs before the code is generated. Therefore we have 2 options:

1. leave in the check for `is_type_complete_v`
2. Move the generate common code before we remove loader specific stuff from the specs

Reply:
I've chosen option 2, but I can be convinced either way tbh.